scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java

```diff
@@ -117,62 +174,204 @@ public void destroy() {
         this.window.destroy();
     }
 
-    /**
-     * internal layout creation
-     * 
-     * <pre>
-     * +- Window ------------------+
-     * |+- VLayout ---------------+|
-     * ||+- Label --------+       || <-- error messages
-     * ||+----------------+       ||     when applicable
-     * ||+- Layout --------------+||
-     * |||+- FormPanel ---------+|||
-     * ||||+- VerticalPanel ---+||||
-     * ||||| form fields       ||||| <-- GWT form wrapped
-     * ||||+-------------------+||||     in SmartGWT layout
-     * |||+---------------------+|||
-     * ||+-----------------------+||
-     * ||+- DynamicForm ---------+||     SmartGWT form, check
-     * ||| form fields           ||| <-- to enable variable edition 
-     * ||+-----------------------+||
-     * ||           +- IButton --+|| <-- submit button
-     * ||           +------------+||
-     * |+-------------------------+|
-     * +---------------------------+
-     * </pre>
-     * 
-     * If the <code>Edit variables</code> checkbox is checked,
-     * the {@link UploadServlet} called by the GWT form will return the content
-     * of the job descriptor, and we will create a new form to edit the
-     * variables so that we may submit the job to a second servlet, {@link SubmitEditServlet}.
-     * If the {@link SubmitEditServlet} submission fails, we get back in the same state
-     * as before the first click to Submit
-     * 
-     * 
-     */
-    private void build() {
+    private void initRootPage() {
+        rootPage = new VLayout();
+        rootPage.setMargin(5);
+        rootPage.setWidth100();
+        rootPage.setHeight100();
+    }
+
+    private void initSelectWfPart() {
+        selectWfLayout = new VLayout();
+        selectWfLayout.setGroupTitle("1. Select workflow");
+        selectWfLayout.setIsGroup(true);
+        selectWfLayout.setHeight("130px");
+
+        // This panel changes depending on the selected method of getting a workflow
+        selectWorkflowButtonsPanel = new VerticalPanel();
+        selectWorkflowButtonsPanel.setSpacing(5);
+        selectWorkflowButtonsPanel.setHeight("50px");
+
+        final VerticalPanel getWfMethodsPanel = new VerticalPanel();
+        getWfMethodsPanel.setSpacing(10);
+        getWfMethodsPanel.setHeight("30px");
+        wfMethodsListBox = new ListBox();
+        wfMethodsListBox.addItem(METHOD_INSTRUCTION);
+        wfMethodsListBox.addItem(METHOD_FROM_FILE);
+        wfMethodsListBox.addItem(METHOD_FROM_CATALOG);
+        getWfMethodsPanel.add(wfMethodsListBox);
+
+        wfMethodsListBox.addChangeHandler(new ChangeHandler() {
+            @Override
+            public void onChange(ChangeEvent event) {
+                String selectedMethod = wfMethodsListBox.getSelectedValue();
+                if (METHOD_FROM_FILE.compareTo(selectedMethod) == 0) {
+                    clearPanel();
+                    initSelectWorkflowFromFilePanel();
```

Comment:
METHOD FROM FILE and FROM CATALOG almost do the same thing : 

clearPanel();
initSelectWorkflowFromCatalogPanel();
...
...
rootPage.removeMember(startAtLayout);
rootPage.removeMember(submitCancelButtons);
initFooter();

A private method to group the common steps?